ical/parsing/property.py

```diff
@@ -88,6 +90,39 @@ def ics(self) -> str:
         result.append(str(self.value))
         return "".join(result)
 
+    @classmethod
+    def from_basic_ics(cls, contentline: str) -> "ParsedProperty":
+        """Decode a ParsedProperty from an rfc5545 iCalendar content line.
+
+        This does not support the full rfc5545 specification. This may only
+        be used for narrower use cases that need more performance than invoking
+        the full rfc5545 spec from the `parser` library.
+
+        Will raise a ValueError on failure.
+        """
+        # Lines can be in either of these formats where the parameters are optional:
+        # RRULE:FREQ=WEEKLY;COUNT=10
+        # RDATE;VALUE=DATE:19970304T080000Z
+        # RDATE:19970304T080000Z
+        if contentline.find(":") == -1:
+            raise ValueError(f"Expected ':' in contentline: {contentline}")
+        name, value = contentline.split(":", 1)
```

Comment:
Use partition instead of split. Now it searches the string twice.
name, sep, value = contentline.partition(":")
if sep is None:
  raise ..

Reply:
Fixed, thanks!

Reply:
You forgot to remove the old calculation of name and value.

Reply:
Yes, saw this, and fixing. also the sep is None check doesn't work. adding tests and coverage still.

Reply:
Ah right it give empty string not None.

Reply:
OK this should be cleaned up. Want to take another look?

Reply:
I'm going to merge but feel free to follow up with any additional feedback.